ui/src/app/components/webcomponents/action-step-form.component.ts

```diff
@@ -1223,33 +1223,34 @@ export class ActionStepFormComponent extends BaseComponent implements OnInit {
   private assignEditTemplate() {
     let firstDataFunction = 0;
     let argumentsList;
-    if(this.testStep?.addonTDF?.value) {
+    if(this.testStep?.addonActionId) {
+      this.showTestDataCF(this.testStep.addonActionId);
       Object.keys(this.testStep?.addonTDF?.value)?.forEach((testdataName, index) => {
         if(index == 0 && this.testStep.addonTDF?.value?.[testdataName]?.type == TestDataType.function){
           firstDataFunction = this.testStep.addonTDF?.value?.[testdataName]?.testDataFunctionId;
           argumentsList = this.testStep.addonTDF?.value?.[testdataName]?.testDataFunctionArguments;
         }
       })
     }
-    if(!!this.testStep?.dataMap?.testData?.size) {
-      this.testDataPlaceholder().forEach((item,index) => {
-        this.testStep.dataMap.testData.forEach((testData: TestDataMapValue) => {
+    if(this.testStep?.dataMap?.testData["test-data"].type == TestDataType.function) {
+      this.showTestDataCF(this.testStep?.testDataFunctionId);
+      this.testStep.dataMap.testData.forEach((testData: TestDataMapValue) => {
```

Comment:
@nixonakash01 You are not assigning testDataFunctionArgs for change in values. That is why there is updation is failing. Have a look at. the payload for update operation being sent to backend

Reply:
there is no testDataFunctionArgs in payload